cxx/pclean/io.cc

```diff
@@ -0,0 +1,284 @@
+
```

Comment:
wdyt of using absl::Status here? https://abseil.io/docs/cpp/guides/status

Reply:
Mildly against.  The absl::Status codes are all built around RPC errors (https://abseil.io/docs/cpp/guides/status-codes), so they wouldn't really help us distinguish between different parsing errors.  Nor would our expected use cases need to so distinguish.  Plus using the abseil libraries would be another piece of overhead to maintain.